src/webview.cpp

```diff
@@ -40,6 +40,19 @@ void WebViewForwardMenu::showEvent(QShowEvent *)
     setGeometry(geo);
 }
 
+void WebView::applyCorrectZoomFactor() {
+    auto url = this->url();
+    auto settingsManager = KiwixApp::instance()->getSettingsManager();
+    qreal zoomFactor;
+    QUrlQuery query(url);
+    if (query.hasQueryItem("pattern")) {
+        zoomFactor = settingsManager->getZoomFactor();
+    } else {
+        auto zimId = url.host().split('.')[0];
```

Comment:
Please follow @mgautierfr's hint in his [earlier comment](https://github.com/kiwix/kiwix-desktop/pull/774#discussion_r803707578)

> Extracting the zimId from the url could (should) be move to a dedicated function